tools/checky/src/checks/mod_use_order.rs

```diff
@@ -0,0 +1,184 @@
+use std::{cmp::Ordering, path::Path};
+
+use proc_macro2::LineColumn;
+use syn::{spanned::Spanned, File, Item, ItemMod, ItemUse, UsePath, UseTree, Visibility};
+
+use crate::output::output_diagnostic;
+
+pub fn check<P>(file_path: P, buffer: &str, file: &File) -> bool
+where
+    P: AsRef<Path>,
+{
+    let mut state = None;
+    for item in file.items.iter() {
+        match item {
```

Comment:
please move the individual variant matches to separate functions to increase readability. Currently, this is extremely nested and convoluted.